### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by natestemen**
## Description

Some random documentation cleanup. Notably, contains the following changes.

1. Try to use more MyST syntax where possible to enhance the docs
2. remove `make doctest` target

### Reviewers

This PR is likely best reviewed commit by commit. I've tried to keep them small for this reason. Sorry there are a few repeated commits for fixing the same issue in multiple files. I thought it would still be easier to review than spend the time rearranging commits and merging them locally.

### CODE REVIEW

File: docs/source/examples/braket_mirror_circuit.md [hunk 1]
```diff
@@ -59,9 +65,14 @@ on_aws = aws_device.name != "DensityMatrixSimulator"
 (examples/braket_mirror_circuit/define-the-circuit)=
 ## Define the circuit
 
-We use mirror circuits to benchmark the performance of the device. Mirror circuits, introduced in https://arxiv.org/abs/2008.11294, are designed such that only one bitstring should be sampled. When run on a device, any other measured bitstrings are due to noise. The frequency of the correct bitstring is our target metric.
+We use mirror circuits to benchmark the performance of the device. Mirror circuits, introduced in <https://arxiv.org/abs/2008.11294>, are designed such that only one bitstring should be sampled. When run on a device, any other measured bitstrings are due to noise. The frequency of the correct bitstring is our target metric.
```
Comment (nathanshammah) [inline docs/source/examples/braket_mirror_circuit.md:68]:
```suggestion
We use mirror circuits to benchmark the performance of the device. Mirror circuits, introduced in {cite}`Proctor_2021_NatPhys`, are designed such that only one bitstring should be sampled. When run on a device, any other measured bitstrings are due to noise. The frequency of the correct bitstring is our target metric.
```

File: mitiq/ddd/rules/rules.py [hunk 2]
```diff
@@ -33,17 +33,22 @@ def general_rule(
...
-            ──I──I──X──I──I──X──I──I──
-            given slack_length = 9, gates [X, Y, X, Y] the spacing defaults
-            to 1 and the rule returns the sequence:
-            ──I──X──I──Y──I──X──I──Y──I──.
         gates: A list of single qubit Cirq gates to build the rule. E.g. [X, X]
             is the xx sequence, [X, Y, X, Y] is the xyxy sequence.
             - Note: To repeat the sequence, specify a repeated gateset.
     Returns:
         A digital dynamical decoupling sequence, as a Cirq circuit.
+
```
Comment (nathanshammah) [inline mitiq/ddd/rules/rules.py:41]: nice

**Review by nathanshammah — APPROVED** [review]
Thanks @natestemen! All renders nicely and better, I just proposed two small improvements on references.

File: docs/source/examples/bqskit.md [hunk 3]
```diff
@@ -193,23 +193,23 @@ Because compiling many large circuits is computationally expensive, we leave the
...
-For more information check out the [`bqskit`](https://bqskit.readthedocs.io/en/latest/) and [`mitiq`](https://mitiq.readthedocs.io/en/stable/) documentation.
+For more information check out the [BQSKit](https://bqskit.readthedocs.io/en/latest/) and [Mitiq](../index.md) documentation.
 
 +++
 
 ### References
 
 - BQSKit documentation: <https://bqskit.readthedocs.io/>
-- BQSKit whitepaper: https://doi.org/10.1145/3503222.3507739
+- BQSKit whitepaper: <https://doi.org/10.1145/3503222.3507739>
```
[the remark this answers is not in the document]
Comment (natestemen) [reply docs/source/examples/bqskit.md]: done in https://github.com/unitaryfund/mitiq/pull/2008/commits/d58082ce17c376b501942ae2b511e73f48f8b6e0

File: docs/source/examples/braket_mirror_circuit.md (hunk 1 above)
Comment (natestemen) [reply docs/source/examples/braket_mirror_circuit.md]: Done in https://github.com/unitaryfund/mitiq/pull/2008/commits/1caa9a72880c82dcbae5c6f497bf199decbaa74f

**Pull request merged**